src/Lotto.js

```diff
@@ -1,18 +1,111 @@
+  constructor() {
+    this.#numbers;
+  }
+
+  setWinningNumbers(numbers) {
     this.validate(numbers);
     this.#numbers = numbers;
   }
 
   validate(numbers) {
-    if (numbers.length !== 6) {
-      throw new Error("[ERROR] 로또 번호는 6개여야 합니다.");
+    if (this.#invalidLength(numbers))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.COUNT}`);
+    if (this.#duplication(numbers))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.DUPLICATION}`);
+    if (isNaN(numbers.join('')))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.NUMBER_ONLY}`);
+    if (this.#invalidNumbersRange(numbers))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.RANGE}`);
+  }
+
+  #invalidLength(numbers) {
+    return numbers.length !== LOTTO.COUNT;
+  }
+
+  #duplication(numbers) {
+    return new Set(numbers).size !== numbers.length;
+  }
+
+  #invalidNumbersRange(numbers) {
+    return (
+      numbers.filter((number) => this.#inRange(number)).length !==
+      numbers.length
+    );
+  }
+
+  #inRange(number) {
+    return LOTTO.RANGE_MIN <= number && LOTTO.RANGE_MAX >= number;
+  }
+
+  addBonusNumber(bonusNumber) {
+    const numTypeBonusNumber = Number(bonusNumber);
+    this.#validateBonusNumber(numTypeBonusNumber);
+    this.#numbers.push(numTypeBonusNumber);
+  }
+
+  #validateBonusNumber(bonusNumber) {
+    if (this.#numbers.includes(bonusNumber))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.DUPLICATION_BONUS}`);
+    if (isNaN(bonusNumber))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.NUMBER_ONLY}`);
+    if (!this.#inRange(bonusNumber))
+      throw new Error(`${ERROR.PREFIX} ${ERROR.RANGE}`);
+  }
+
+  getMatchCount(oneLottoNumbers) {
+    const winningNumbersDeletedBonus = this.#numbers.slice(0, LOTTO.COUNT);
+
+    return oneLottoNumbers.filter((lottoNumber) =>
+      winningNumbersDeletedBonus.includes(lottoNumber)
+    ).length;
+  }
+
+  #isBonus(oneLottoNumbers) {
+    const bonusNumber = this.#numbers[LOTTO.COUNT];
+    const matchCount = this.getMatchCount(oneLottoNumbers);
+    return (
+      oneLottoNumbers.includes(bonusNumber) && matchCount === LOTTO.COUNT - 1
+    );
+  }
+
+  initLottoResultStorage() {
+    const storage = {};
+    for (let i = LOTTO.MATCH_START_COUNT; i <= LOTTO.COUNT; i++) {
+      storage[`${SENTENCE.MATCH_COUNT(i)}`] = 0;
+      if (i === LOTTO.COUNT - 1) {
+        // 보너스 볼인 경우
+        storage[`${SENTENCE.MATCH_COUNT(i)}, ${SENTENCE.MATCH_BONUS}`] = 0;
+      }
     }
+    return storage;
   }
 
-  // TODO: 추가 기능 구현
+  getLottoResult(allLottoNumbers) {
+    const lottoResult = this.initLottoResultStorage();
+
+    allLottoNumbers.forEach((oneLottoNumbers) => {
+      const matchCount = this.getMatchCount(oneLottoNumbers);
+      if (this.#isBonus(oneLottoNumbers))
+        lottoResult[
+          `${SENTENCE.MATCH_COUNT(matchCount)}, ${SENTENCE.MATCH_BONUS}`
+        ]++;
+      else if (matchCount >= LOTTO.MATCH_START_COUNT)
+        lottoResult[`${SENTENCE.MATCH_COUNT(matchCount)}`]++;
+    });
```

Comment:
다음과 같은 구문에서 문장이 길어진다 싶으면 가독성을 위해 상수를 구조 분해 할당 해도 좋을 것 같아요!

```suggestion
    const { MATCH_COUNT, MATCH_BONUS } = SENTENS;
    allLottoNumbers.forEach((oneLottoNumbers) => {
      const matchCount = this.getMatchCount(oneLottoNumbers);
      if (this.#isBonus(oneLottoNumbers))
        lottoResult[`${MATCH_COUNT(matchCount)}, ${MATCH_BONUS}`]++;
      else if (matchCount >= LOTTO.MATCH_START_COUNT)
        lottoResult[`${MATCH_COUNT(matchCount)}`]++;
    });
```

추가적으로 `++` 연산자보다는 `+= `를 활용해 1을 더해주는 방법이 더 좋을 것 같아요!
이 부분은 js 컨벤션에 있습니다!

https://github.com/airbnb/javascript#variables--unary-increment-decrement

Reply:
와 이건 정말 작성하면서도 상수참조가 너무 길어서 어찌 해결하는지 방법을 몰랐는데 이런 방법이 있었네요!! 컨벤션을 분명 다 읽었다고 생각했는데 아니었네요 ㅠㅜ 이런 부분 알려주셔서 정말 감사합니다!! 컨벤션을 한번 더 읽어봐야 겠어요😱